servicex_app/servicex/code_gen_adapter.py

```
@@ -36,28 +36,52 @@ def __init__(self, code_gen_url, transformer_manager):
             raise ValueError(f'Failed to generate translation code: {msg}')
 
-        zipfile = ZipFile(BytesIO(result.content))
+        # Multipart Mime Encoder from requests_toolbelt appends a delimited between all the fields mentioned in the data payload
+        # Here we are using two constants START_INDEX=2 and END_INDEX=34 which will extract the delimiter from the string
+        # The delimiter is used by the decoder to extract the data out the individual fields from the multipart data
+
+        START_INDEX = 2
+        END_INDEX = 34
+        boundary = result.data[START_INDEX:END_INDEX].decode('utf-8')
+        content_type = f"multipart/form-data; boundary={boundary}"
+        decoder_parts = decoder.MultipartDecoder(result.data, content_type)
 
         assert self.transformer_manager, "Code Generator won't work without a Transformer Manager"
-        return self.transformer_manager.create_configmap_from_zip(zipfile,
-                                                                  request_record.request_id,
-                                                                  namespace)
+
+        if len(decoder_parts.parts) > 1:
+            transformer_image = str(decoder_parts.parts[0].content, 'utf-8')
+            zipfile = decoder_parts.parts[1].content
+
+            zipfile = ZipFile(BytesIO(zipfile))
+
+            return (self.transformer_manager.create_configmap_from_zip(zipfile,
+                                                                       request_record.request_id,
+                                                                       namespace), transformer_image)
+        else:
+            zipfile = ZipFile(BytesIO(""))
+            return (self.transformer_manager.create_configmap_from_zip(zipfile,
+                                                                       request_record.request_id,
+                                                                       namespace), transformer_image)
```

I wonder if we should report an error if `decoder_parts.parts` <> 2

Certainly `transformer_image` is undefined here

---

I don't think it would happen, I had just put up this condition for the test case to run, incase the codegen lib fails, it would return a code 500, otherwise there is no reason for it to return decoder_parts < 2

---

I have removed the check for decoder_parts len < 1 please do check the new version, all test cases are working fine on  it.